### PR TITLE
[MM-61140] Allow plugins to add Support Packet data without UI elements

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
https://github.com/mattermost/mattermost/pull/26403 added the possibility for plugins to extend the Support Packet.

For https://mattermost.atlassian.net/wiki/spaces/CLOUD/pages/3083108354/Diagnostics+data+from+Core+Plugins+to+the+Support+Packet, we want to extend the Support Packet with plugin data by default. To enable easier debugging we want to always include the Support packet information from these plugins.

This PR introduces the possibility for plugins to add Support Packet data without a toggle in the UI. That is done by checking if the plugin defines the `support_packet` (which controls the text in the UI).

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-61140

#### Release Note
```release-note
Allow plugins to add Support Packet data without UI elements
```

### CODE REVIEW

File: server/channels/app/support_packet.go [hunk 1]
```diff
@@ -82,26 +84,30 @@ func (a *App) GenerateSupportPacket(c request.CTX, options *model.SupportPacketO
...
-			for _, data := range pluginData {
-				fileDatas = append(fileDatas, *data)
+	pluginContext := pluginContext(c)
+	a.ch.RunMultiHook(func(hooks plugin.Hooks, manifest *model.Manifest) bool {
+		// If the plugin defined the support_packet prop it means there is a UI element to include it in the support packet.
+		// Check if the plugin is in the list of plugins to include in the Support Packet.
+		if _, ok := manifest.Props["support_packet"]; ok {
+			if !slices.Contains(options.PluginPackets, manifest.Id) {
+				return false
 			}
```
Comment (isacikgoz) [inline server/channels/app/support_packet.go:94]: Could this simplified to: 
```suggestion
			return slices.Contains(options.PluginPackets, manifest.Id)
```

File: server/channels/app/user.go [hunk 2]
```diff
@@ -1045,10 +1045,10 @@ func (a *App) UpdateActive(c request.CTX, user *model.User, active bool) (*model
 
 	a.sendUpdatedUserEvent(*ruser)
 
-	if pluginsEnvironment := a.GetPluginsEnvironment(); pluginsEnvironment != nil && !active && user.DeleteAt != 0 {
+	if !active && user.DeleteAt != 0 {
```
Comment (isacikgoz) [inline server/channels/app/user.go:1048]: Why we gave up on `pluginsEnvironment` check here?

**Review by isacikgoz — APPROVED** [review]
LGTM, a couple of non-blocking things.

**Review by BenCookie95 — APPROVED** [review]
LGTM, I think it would be a good idea to add an example of this to the plugin starter template

**Comment by hanzei** [comment]
@BenCookie95 Thanks for the suggestion! There is already an example in the demo plugin (which used the UI): https://github.com/mattermost/mattermost-plugin-demo/pull/175. I think that is sophisticated enough.

File: server/channels/app/user.go (hunk 2 above)
Comment (hanzei) [reply server/channels/app/user.go:1048]: `RunMultiHook` does check if the a pluginsEnvironment exists. This means that with the new code, we do spawn a go routine, even if plugins are disabled. Given that this is consistent with the rest of the code base (e.g. https://github.com/mattermost/mattermost/blob/773ad33369ac4f227a4fff959b5a02c012655d30/server/channels/app/platform/web_conn.go#L269-L275) I think it's fine.

**Review by DHaussermann — APPROVED** [review]
Tested and passed 
I deployed this along with the corresponding Enterprise PR

Confirmed `demo_plugin_support_packet.txt` is now included in the support packet.

LGTM!

**Comment by DHaussermann** [comment]
@hanzei in case it was needed I quickly repeated this testing after the merge into this branch and the related Enterprise PR branch to ensure no changes in behavior. 👍

**Pull request merged**